Homework two

StasOskol opened the pull request:
Реализовал настройку Babel, оптимизацию картинок, mp3 формат.Был прописан порт,для запуска,оптимизация, и очистка при новой сборке.
Источник:https://my-class.ru/kak-nastroit-babel-v-webpack/


Насчёт уменьшения размера сторонних библиотек ,то с музыкой так и не получилось сделать,а luxon не удалось обработать. Смотрел в источнике:https://www.youtube.com/watch?v=XY2NLKCrjJ4,но нужно разбираться как и что работает в самой библиотеке,потом использовал null-loader ,но он снёс все js ,потом в методичке смотрел про tree-shaking, поставил в json-файл,но также как и null всё снёс,webpack я настроил и сборка работает,не получилось уменьшить размер

Вопросы: есть сборка production, но как ее настроить,чтобы и собирался и разработчика и production?
Map я настроил
Сборку проверял - работает 
В luxon пытался взять min-ный формат библиотеки,но вес не пропал

StasOskol committed:
Добавление всех файлов и кода для выполнения пункта домашнего задания

StasOskol committed:
Разбиение на подпункты

StasOskol committed:
Добавление файлов для калькулятора дат

StasOskol committed:
Добавление разметки для калькулятора дат

StasOskol committed:
Модернизация для использования модулей

StasOskol committed:
Проверка на работу дефолтного импорта

StasOskol committed:
Вывод результата разности дат в консоль
Показ подключения библиотеки luxon двумя способами, через сервер и
локально (код)
Использование библиотеки рассчёта
Подключение не дефолтным импортом
Применение функции, лежащей отдельно, но выводящая, через RETURN
результат (остальные рассчёты private)

StasOskol committed:
Вывод результата на страницу
Использование одновременно дефолтного export и именованного

StasOskol committed:
Добавление комментарием пояснений

StasOskol committed:
Выполнение 2 задания подпункт а с массивом
Добавление файлов и зависимостей b от а
Вывод всех скриптов в head

StasOskol committed:
Реализация просто с callback, но с ? (не понятен этот способ)

StasOskol committed:
Решение с зависимостью массива

StasOskol committed:
Реализация в отдельную функцию (проверка)

StasOskol committed:
Выполнено задание на обнаружение повторений
Выделена отдельная функция для проверки
Подключена в разные файлы для проверки на повторяемость
Если нужно добавить новый скрипт - использовать функцию checkReplay

StasOskol committed:
Работа с AMD (RequireJS)

StasOskol committed:
Merge pull request #1 from StasBeep/one-point

One point (Выполнено 2 пункт домашнего задания)

StasOskol committed:
Подведение к реализации 1 домашнего задания

StasOskol committed:
Реализация работы таймера
Написание своего модуля для таймера с выводом на страницы текущих
изменений
Использовано setInterval для отчёта времени

StasOskol committed:
Подпись функции для реализации таймера (подклчение звука)
Подключить библиотеку hower не получилось (не через node.js, не через
github, просто скопировать тоже не получилось)
Включение трека, когда таймер на 0 0 0

StasOskol committed:
Добавление стилей и обработки при не правильности введений данных

StasOskol committed:
Полное (првильное) выполнение работы таймера
Завершение работы с кнопкой стоп
Правльная работа при нажатии старт
Полная модуляция
Добавление функции на выключение звуковой композиции

StasOskol committed:
Выполнено домашнее задание 1 части
Работа с меню (при нажатии на кнопки переключение между калькулятором и
таймаром)
Добавление стилей

StasOskol committed:
Перемещение проекта в сборку, чтобы проще было делать сборку + удаление не нужных файлов для сборки

StasOskol committed:
Обновление ignore, распаковка webpack

StasOskol committed:
Сборка проекта на webpack сделана (babel & оптимизации нет)
1. Добавлены не нужные папки и файлы
2. Настроен webpack.config.js, который работатет со стилями и файлами, а
   также анализирует количество занятого места
3. В index.html убираем не нужные строчки с подключением script и css
   (сборка сделает всё сама)
4. В index.js добавлен импорт для sass (отдельный плагин и библиотека)
5. sound.js добавлен новый импорт, который преобразует путь к mp3-файлу
6. Настроен package.json, который при командах:
	npx run build - собирает проект
	npx run start - собирает и запускает хостинг с отслеживанием
	действий

StasOskol committed:
Merge pull request #3 from StasBeep/work-webpack-classwork

Work webpack classwork (end)

StasOskol committed:
реализация сборки завершена с вопросами

marinatells commented on src/worktimer.js:
```diff
@@ -0,0 +1,115 @@
+
+            // ! Подключение библиотеки howler не получилось
+            /*
+            let sound = new Howl({
```
howler нужно поставить через npm
после чего импортировать её и отдельно импортировать звук 'sound.mp3' ( как делали с картинкой на уроке )
После этого можно будет использовать импортированный звук

marinatells commented on webpack.config.js:
```diff
@@ -0,0 +1,87 @@
+    },
+
+    optimization: {
+        minimizer: [new UglifyJsPlugin({
```
вместо uglify теперь рекомендуется использовать terser https://github.com/webpack-contrib/terser-webpack-plugin

marinatells commented on webpack.config.js:
```diff
@@ -0,0 +1,87 @@
+    // Готовый продукт
+    // mode: 'production',
+    // Сборка для разработки
+    mode: 'development',
```
настроить под дев и бой можно только создав два разных конфига и прописать для их запуска npm скрипты
есть способы сделать один конфиг на основе другого (https://webpack.js.org/guides/production/)

marinatells commented on src/date.js:
```diff
@@ -0,0 +1,16 @@
+// import {DateTime} from 'https://moment.github.io/luxon/es6/luxon.js';
+
+// Второй способ подключения библиотеки
+import {DateTime} from './luxon.js';
```
можно немного снизить размер сборки, если установить luxon через npm и применить к нему тот же способ, что для Lodash — то есть импортировать только тот файл, в котором лежит datetime, а не всю библиотеку
```
import DateTime from 'luxon/src/datetime';
```

StasOskol committed:
Полная и хорошая настройка webpack (пользуйся)
Хорошо настроенный webpack.config.js
Хорошо настроенный .gitignore
Хорошо настроенный package.json
